src/sync/app.cpp

```diff
@@ -622,5 +622,74 @@ void App::log_out(std::function<void (Optional<AppError>)> completion_block) con
     log_out(current_user(), completion_block);
 }
 
+void App::switch_user(std::shared_ptr<SyncUser> user,
+                      std::function<void(std::shared_ptr<SyncUser>, Optional<AppError>)> completion_block) const
+{
+    int sync_user_idx = -1;
+    for(size_t i = 0; i < SyncManager::shared().all_users().size(); i++) {
+        if (SyncManager::shared().all_users()[i] == user) {
+            sync_user_idx = static_cast<int>(i);
+            break;
+        }
+    }
+    
+    if (sync_user_idx == -1) {
+        return completion_block(nullptr,
+                                AppError(make_custom_error_code(ClientErrorCode::user_not_found), "User does not exist"));
+    }
+    
+    auto sync_user = SyncManager::shared().all_users()[sync_user_idx];
+    if (!sync_user->is_logged_in()) {
+        return completion_block(nullptr,
+                                AppError(make_custom_error_code(ClientErrorCode::user_not_logged_in), "User is not longer valid"));
+    }
+    
+    SyncManager::shared().set_current_user(user->identity());
+    
+    return completion_block(current_user(), {});
+}
+
+void App::remove_user(std::shared_ptr<SyncUser> user,
+                      std::function<void(Optional<AppError>)> completion_block) const
+{
+    int sync_user_idx = -1;
+    for(size_t i = 0; i < SyncManager::shared().all_users().size(); i++) {
+        if (SyncManager::shared().all_users()[i] == user) {
+            sync_user_idx = static_cast<int>(i);
+            break;
+        }
+    }
+    
+    if (sync_user_idx == -1) {
+        return completion_block(AppError(make_custom_error_code(ClientErrorCode::user_not_found),
+                                         "No user has been found"));
+    } else {
+        auto sync_user = SyncManager::shared().all_users()[sync_user_idx];
+        if (sync_user->is_logged_in()) {
+            sync_user->log_out();
+        }
+        SyncManager::shared().remove_user(user->identity());
+    }
+
+    return completion_block({});
+}
+
+void App::remove_user(std::function<void(Optional<AppError>)> completion_block) const
```

Comment:
[q] Do we need this function?

Reply:
It is potentially confusing as we do not know who we are removing, removed this function as a result of that